core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java

```diff
@@ -22,22 +22,38 @@
 import java.util.BitSet;
 
 import org.apache.carbondata.core.constants.CarbonCommonConstants;
-import org.apache.carbondata.core.datastore.page.statistics.ColumnPageStatistics;
+import org.apache.carbondata.core.constants.CarbonV3DataFormatConstants;
+import org.apache.carbondata.core.datastore.compression.Compressor;
+import org.apache.carbondata.core.datastore.page.statistics.ColumnPageStatsVO;
 import org.apache.carbondata.core.metadata.datatype.DataType;
+import org.apache.carbondata.core.util.ByteUtil;
+import org.apache.carbondata.core.util.CarbonProperties;
 import org.apache.carbondata.core.util.DataTypeUtil;
 
-// Represent a columnar data in one page for one column.
+import static org.apache.carbondata.core.metadata.datatype.DataType.BYTE;
+import static org.apache.carbondata.core.metadata.datatype.DataType.DECIMAL;
+import static org.apache.carbondata.core.metadata.datatype.DataType.DOUBLE;
+import static org.apache.carbondata.core.metadata.datatype.DataType.FLOAT;
+import static org.apache.carbondata.core.metadata.datatype.DataType.INT;
+import static org.apache.carbondata.core.metadata.datatype.DataType.LONG;
+import static org.apache.carbondata.core.metadata.datatype.DataType.SHORT;
+import static org.apache.carbondata.core.metadata.datatype.DataType.STRING;
+
+/**
+ * Represent a columnar data in one page for one column.
+ */
 public class ColumnPage {
 
-  protected final DataType dataType;
-  protected final int pageSize;
-  protected ColumnPageStatistics stats;
+  private final int pageSize;
+  private DataType dataType;
+  private ColumnPageStatsVO stats;
```

Comment:
I think ColumnPageStats supposed to be interface so that user can implement to have his own stats